src/rust/engine/fs/store/src/lib.rs

```diff
@@ -598,6 +597,48 @@ impl Store {
       .to_boxed()
   }
 
+  /// Load a REv2 Tree from a remote CAS and cache the embedded Directory protos in the
+  /// local store. Tree is used by the REv2 protocol as an optimization for encoding the
+  /// the Directory protos that compromose the output directories from a remote
+  /// execution reported by an ActionResult.
+  ///
+  /// Returns an Option<Digest> representing the `root` Directory within the Tree (if it
+  /// in fact exists in the remote CAS).
+  ///
+  /// This method requires that this Store be configured with a remote CAS (and will return
+  /// an error if this is not the case).
+  pub async fn load_tree_from_remote(&self, tree_digest: Digest) -> Result<Option<Digest>, String> {
+    let remote = if let Some(ref remote) = self.remote {
+      remote
+    } else {
+      return Err("Cannot load Trees from a remote without a remote".to_owned());
+    };
+
+    let tree_opt = remote
+      .load_bytes_with(EntryType::Directory, tree_digest, |b| {
```

Comment:
This should probably be `EntryType::File` which is nominally less constraining, though as you noted before this arg is maybe actually ignored?

Reply:
For the remote `Store` it is ignored. I'm fine with passing either variant here. Would `EntryType::File` be preferred then?

Reply:
Honestly let's just remove the param - there's no reason for it to exist (IIRC the `local::ByteStore` and `remote::ByteStore` used to implement the same trait, but they don't any more), so let's avoid needing to even ask the question :)

Reply:
Will do.